frontend/components/Navbar.tsx

```diff
@@ -45,37 +57,40 @@ function WrappedNavLink({ title, path, as, reactive }) {
   );
 
   if (reactive) {
-    return (
-      <Link href={path} as={as}>
-        {navLink}
-      </Link>
-    );
+    return <Link href={path}>{navLink}</Link>;
   } else {
     return navLink;
   }
+};
+
+interface WrappedNavDropdownProps {
+  title: string;
+  submenuItems: Omit<GlobalMenuItem, "submenuItems">[];
 }
 
-function WrappedNavDropdown({ title, children, ...childProps }) {
+const WrappedNavDropdown: FC<WrappedNavDropdownProps> = ({
+  title,
+  submenuItems,
+}: WrappedNavDropdownProps) => {
   return (
-    <NavDropdown renderMenuOnMount title={title} {...childProps}>
-      {children.map((item) => (
+    <NavDropdown renderMenuOnMount title={title} id={`${title}-dropdown`}>
+      {submenuItems.map((item) => (
         <WrappedNavLink key={item.path} {...item} />
       ))}
     </NavDropdown>
   );
-}
-
-// https://reactjs.org/docs/typechecking-with-proptypes.html
-WrappedNavDropdown.propTypes = {
-  title: PropTypes.string.isRequired,
-  children: PropTypes.arrayOf(PropTypes.object),
 };
 
-export default function GlobalNavbar({ menuItems }) {
+interface GlobalNavbarProps {
+  menuItems?: GlobalMenuItem[];
+}
+
+const GlobalNavbar: FC<GlobalNavbarProps> = ({ menuItems }: GlobalNavbarProps) => {
   menuItems = menuItems || globalMenuItems;
 
   const router = useRouter();
-  const [session, loading] = useSession();
+  // TODO: Create session type and remove this cast to `any`.
+  const [session, loading] = useSession() as any;
```

Comment:
Casting for now, since otherwise we get complaints when trying read attributes like `session.user["avatar"]`.